### ISSUE TITLE
Better example/support for reading lines from a file

### ISSUE THREAD

**Issue by Miserlou** [Miserlou/Loop#12]
This command gives weird behavior:

```
cargo run "echo $ITEM" --for "`cat README.md`"
```

It'd be great to have a better example for this use case, either a) fixing this behavior, b) explicitly adding the ability to read in a file from stdin c) adding explicit reading of a file or d) all of the above.

**Comment by madeddie** [comment]
What weird behaviour are you seeing? Empty lines?
If so, try:
```cargo run 'echo $ITEM' --for "`cat README.md`"```

**Comment by Miserlou** [comment]
Yes, I was seeing empty lines, but that fixed it. Why does the type of quote matter?

**Comment by madeddie** [comment]
That's because of your shell, probably bash.

Try `echo "$TERM"` and then `echo '$TERM'`

You'll see that the first time it outputs the value of $TERM, but the second time it outputs a literal $TERM. Double quotes allow variable substitution, single quotes prohibit.

Specifically, when you execute cargo run, the $ITEM is replaced with the value of $ITEM in the parent shell of cargo, aka, the current shell you're in, instead of the shell that loop spins up. $ITEM is most likely empty there.

The same goes for the value of \`cat README.md\`. This last one doesn't cause a problem though, since both outside loop and inside loop, \`cat README.md\` would have the same output.

Does that make sense?

**Comment by madeddie** [comment]
For more info, see for instance http://wiki.bash-hackers.org/syntax/quoting

mind you, this also goes with other advanced shell like ksh.

**Comment by Miserlou** [comment]
Ha! Okay, thanks for explaining! I learned something new today.

**Comment by michaelmior** [comment]
Looks like this can be closed?
